meshpy/four_c/beam_to_beam_contact.py

```diff
@@ -0,0 +1,65 @@
+# -*- coding: utf-8 -*-
+# -----------------------------------------------------------------------------
+# MeshPy: A beam finite element input generator
+#
+# MIT License
+#
+# Copyright (c) 2018-2024
+#     Ivo Steinbrecher
+#     Institute for Mathematics and Computer-Based Simulation
+#     Universitaet der Bundeswehr Muenchen
+#     https://www.unibw.de/imcs-en
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+# -----------------------------------------------------------------------------
+"""This file contains the definition of the beam to beam contact condtion for
+4c."""
+
+# MeshPy modules
+from ..boundary_condition import BoundaryCondition
+from ..conf import mpy
+from ..geometry_set import GeometrySet
+
+
+class BeamtoBeamContactCondition(BoundaryCondition):
+    """This object represents the beam to beam contact condition in 4C.
+
+    It sets the conditions to enable contact between beams.
+    """
+
```

Comment:
Here I have a question:
Currently, I have to add two times the same Boundary Condition.
Wouldn't it make more sense to have something like:
`BeamtoBeamContactCondition(nodes1,nodes2)` 
and under the hood the function create the boundary conditions for `nodes1` and `nodes2` and keeps track automatically of the IDs ?

Reply:
Per definition a boundary condition is always something that acts on a single geometry set in MeshPy. So it is not possible to add two geometry sets to a single BC. What is however possible is to have a factory function or object (similar to what you proposed) what takes `beam_contact_bc_factory(mesh, nodes1, nodes2, bc_description, ...)` and creates the corresponding BCs automatically. The ID count could either be stored in the factory object, a global variable, or be retrieved by checking how many `beam_to_beam_contact` conditions are already in the mesh and dividing that number by 2. @knarfnitram @davidrudlstorfer do you have an idea what would be a Pythonic way to track the number of added IDs? I am not sure if global variables and objects are desirable.